Sources/DataConnectError.swift

```diff
@@ -14,27 +14,239 @@
 
 import Foundation
 
-/// Represents an error returned by the DataConnect service
+// MARK: Base Error Definitions
+
+/// Protocol representing an error returned by the DataConnect service
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+public protocol DataConnectError: Error, CustomDebugStringConvertible, CustomStringConvertible {
+  var message: String? { get }
+  var cause: Error? { get }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+public extension DataConnectError {
+  var debugDescription: String {
+    return "{\(Self.self), message: \(message ?? "nil"), cause: \(String(describing: cause))}"
+  }
+
+  var description: String {
+    return debugDescription
+  }
+}
+
+/// Type erased DataConnectError
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
-public enum DataConnectError: Error {
-  /// no firebase app specified. configure not complete
-  case appNotConfigured
+public struct AnyDataConnectError: DataConnectError {
```

Comment:
I forget, why is this `AnyDataConnectError` type needed?

Looking at the usage, can we not use the `DataConnectError`?
```diff
- private let resultsPublisher = PassthroughSubject<Result<ResultData, AnyDataConnectError>,
+ private let resultsPublisher = PassthroughSubject<Result<ResultData, DataConnectError>,
```

Reply:
The Publisher generic definition isn't accepting an inherited protocol of Error. It needs a conrete type. So I created this type erased one.

Reply:
Actually wait. I think I just need to declare the associatedtype for DataConnectError in the QueryRef protocol.